### PR TITLE
Add types to project

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Now that all of our code is typed, TypeScript can check for issues.

This doesn't enable strict mode because that requires even more changes. It may be worth doing that in a follow up.

### CODE REVIEW

File: src/js/about.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* global document, window */
 const setUpAbout = () => {
-  const aboutElement = document.querySelector(".about-popup");
+  const aboutElement = document.querySelector(".about-popup") as HTMLElement;
```
Comment (Eric-Arellano) [inline src/js/about.ts:3]: FYI `as HTMLElement` is telling TypeScript "trust me, this is this type". An alternative would be to check with runtime code that it's an `HTMLElement` and error if so. That would be more robust. But it's more boilerplate, and we already have tests to check that this works properly.

File: src/js/filter.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* global document, window */
```
Comment (Eric-Arellano) [inline src/js/filter.ts:1]: I didn't make changes to this file - only added types. But the changes are enough that Git thinks it's a new file.

File: src/js/filter.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const changeSelectedMarkers = (
+  markerGroup: FeatureGroup,
+  citiesToMarkers: Record<CityId, CircleMarker>,
+  filterFn: (cityState: CityId) => boolean
+) => {
+  Object.entries(citiesToMarkers).forEach(([cityState, marker]) => {
+    if (filterFn(cityState)) {
+      marker.addTo(markerGroup);
+    } else {
+      // @ts-ignore the API allows passing a LayerGroup, but the type hint doesn't show this.
```
Comment (Eric-Arellano) [inline src/js/filter.ts:19]: FYI this is new

File: src/js/filter.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  (
+    markerGroup: FeatureGroup,
+    citiesToMarkers: Record<CityId, CircleMarker>,
+    data: Record<CityId, CityEntry>
+  ) =>
+  (): void => {
+    const selected = new Set(
+      Array.from(document.querySelectorAll(".filter--scope :checked")).map(
+        (option: HTMLInputElement) => option.value
+      )
```
Comment (Eric-Arellano) [inline src/js/filter.ts:36]: This was a change. We used ot use `[...document.querySelectorALl()]` to create an array. It's better to use `Array.from()`.

File: src/js/setUpSite.ts [hunk 5]
```diff
@@ -17,7 +18,6 @@ const BASE_LAYER = new TileLayer(
     subdomains: "abcd",
     minZoom: 0,
     maxZoom: 10,
-    ext: "png",
```
Comment (Eric-Arellano) [inline src/js/setUpSite.ts:20]: This isn't valid to set.

File: src/js/setUpSite.ts [hunk 6]
```diff
@@ -52,9 +52,10 @@ const createMarkerGroup = (map) => {
 /**
  * Read the CSV and return an object with `City, State` as the key and the original entry as the value.
  */
-const readData = async () => {
+const readData = async (): Promise<Record<CityId, CityEntry>> => {
+  // @ts-ignore
   const data = await import("../../map/tidied_map_data.csv");
```
Comment (Eric-Arellano) [inline src/js/setUpSite.ts:57]: TypeScript doesn't know how the CSV is imported

**Review by tunglinn — APPROVED** [review]

**Pull request merged**